### PR TITLE
Improve code quality of Ui Components

### PR DESCRIPTION
**Pull request by praveenkrishna0512**

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -93,21 +96,24 @@ public void updateFilteredPatientList(Predicate<Patient> predicate) {
...
-            optionalDoctor = Optional.of(model.getFilteredDoctorList().get(0));
-        }
-        return optionalDoctor;
+    public void updateFilteredPatientListBasedOnDoctor(Doctor doctor) {
+        requireNonNull(doctor);
+        PatientFilter patientContainsDoctor =
+                new PatientFilter(doctor);
+        Predicate<Patient> patientsOfDoctorPredicate =
+                new PatientContainsKeywordsPredicate(patientContainsDoctor);
+        this.updateFilteredPatientList(patientsOfDoctorPredicate);
```
Comment (owen-yap) [inline src/main/java/seedu/address/logic/LogicManager.java:105]: Is there any reason for replacing getDoctorIfPresent with updateFilteredPatientList?

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 2]
```diff
@@ -13,104 +13,103 @@
...
 
     /** The application should exit. */
-    private final boolean exit;
+    private final boolean shouldExit;
 
     /** This command interacts directly with the GUI */
     private final boolean hasGuiInteraction;
 
-    /** Doctor selected by user through command */
-    private final Optional<Doctor> selectedDoctor;
```
Comment (owen-yap) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:28]: Is there any reason for making this change?

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 3]
```diff
@@ -13,104 +13,103 @@
...
 
     /**
      * Constructs a {@code CommandResult} with selected patient.
      */
-    public CommandResult(String feedbackToUser, boolean hasGuiInteraction, Patient selectedPatient) {
-        this(feedbackToUser, false, false, hasGuiInteraction, null, selectedPatient);
+    public CommandResult(String cliFeedbackToUser,
+                         Patient selectedPatient) {
+        this(cliFeedbackToUser, false, false,
+                true, null, selectedPatient);
```
Comment (owen-yap) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:87]: Should we set defaults instead of calling the main constructor with boolean values?

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 4]
```diff
@@ -73,7 +73,7 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook) throws IOException
     @Override
     public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
         logger.fine("Attempting to write to data file: " + filePath);
-        logger.log(Level.WARNING, "Updating the file manually may result in unexpected behaviour and/or data loss");
+        logger.log(Level.INFO, "Updating the file manually may result in unexpected behaviour and/or data loss");
```
Comment (owen-yap) [inline src/main/java/seedu/address/storage/StorageManager.java:76]: Why this change? What is the difference here?

**Review by owen-yap — COMMENTED** [review]
Looks good! Thanks for the hard work. I left some comments, mostly just wanted to ask some things to clarify.

Be careful not to go too deep into refactoring code just for the sake of it. If it follows the coding convention and basic OOP principles, we should be good to go.

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 1 above)
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/logic/LogicManager.java:105]: It's not a replacement; the `getDoctorIfPresent` function is not being used

File: src/main/java/seedu/address/logic/commands/CommandResult.java (hunk 2 above)
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:28]: Yeap! Realised that it is not the intended use of Optionals to be parameter types. Rather, they should be used as return types to indicate potential absense of a value.

File: src/main/java/seedu/address/logic/commands/CommandResult.java (hunk 3 above)
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:87]: Hmm you are right! let me make the changes

File: src/main/java/seedu/address/storage/StorageManager.java (hunk 4 above)
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/storage/StorageManager.java:76]: This was adjusted to not cause alarm to any developers when looking through logs, as this function call happens quite often. Furthermore, the string acts as more of a tip rather than a warning at this specific point as the user interaction does not have anything to do with the warning itself.

**Review by vigonometry — COMMENTED** [review]
LGTM, bear in mind what Owen says about not getting too deep into refactoring

**Review by vigonometry — APPROVED** [review]
LGTM
